scripts/superdesk-search/react.js

```diff
@@ -1576,6 +1588,35 @@ import classNames from 'classnames';
                             }, 500, false);
                         },
 
+                        selectItem: function (item) {
+                            if (isCheckAllowed(item)) {
+                                var selected = !item.selected;
+                                this.multiSelect([item], selected);
+                            }
+                        },
+
+                        selectMultipleItems: function (lastItem) {
+                            var itemId = search.generateTrackByIdentifier(lastItem),
+                                positionStart = 0,
+                                positionEnd = _.indexOf(this.state.itemsList, itemId),
+                                selectedItems = [];
+
+                            if (this.state.selected) {
+                                positionStart = _.indexOf(this.state.itemsList, this.state.selected);
+                            }
+
+                            for (var i = positionStart; i <= positionEnd; i++) {
+                                var item = this.state.itemsById[this.state.itemsList[i]];
+                                if (isCheckAllowed(item)) {
+                                    selectedItems.push(item);
+                                }
+                            }
+
+                            multi.reset();
```

Comment:
It works same as in ubuntu, it will select from last item you picked. For example, with ctrl you select item 1, 2 and 3, then press shift and select 6th item. Selection will be from 3 to 6 (as I say, same as in windows/ubuntu file system)